File: Latihan_B/python/Pc.py
from Processor import Processor
from Disk import Disk
from Ram import Ram

class Pc:
    # atribut private
    __objProcie = Processor()
    __objStorage = Disk()
    __objMemory = Ram()
    __totalPrice = 0

    # constructor
    def __init__(self, procie, storage, memory, totalPrice=0):
        self.__objProcie = procie
        self.__objStorage = storage
        self.__objMemory = memory
        self.__totalPrice = totalPrice

    # method set dan get untuk procie, storage, memory, dan totalPrice
    def setProcie(self, procie):
        self.__objProcie = procie
    def setStorage(self, storage):
        self.__objStorage = storage
    def setMemory(self, memory):
        self.__objMemory = memory
    def setPrice(self, totalPrice):
        self.__totalPrice = totalPrice

    def getProcie(self):
        return self.__objProcie
    def getStorage(self):
        return self.__objStorage
    def getMemory(self):
        return self.__objMemory
    def getTotalPrice(self):
        return self.__totalPrice
    
    # method print
    def printComputer(self):
        self.__objProcie.printProcie()
        self.__objStorage.printDisk()
        self.__objMemory.printRam()
        print("Total Price : " + "Rp. " + str(self.getTotalPrice()))

File: Latihan_B/java/Pc.java
public class Pc
{
    private Processor objProc;
    private Disk objStorage;
    private Ram objMemory;
    private int totalPrice;

    // constructor
    public Pc(){}
    public Pc(Processor objProc, Disk objStorage, Ram objMemory, int totalPrice) {
        this.objProc = objProc;
        this.objStorage = objStorage;
        this.objMemory = objMemory;
        this.totalPrice = totalPrice;
    }

    // method set dan get untuk objProc, objStorage, objMemory, dan totalPrice
    public void setProcie(Processor objProc) {
        this.objProc = objProc;
    }
    public void setStorage(Disk objStorage) {
        this.objStorage = objStorage;
    }
    public void setMemory(Ram objMemory) {
        this.objMemory = objMemory;
    }
    public void setTotalPrice(int totalPrice) {
        this.totalPrice = totalPrice;
    }

    public Processor getProcie() {
        return objProc;
    }
    public Disk getStorage() {
        return objStorage;
    }
    public Ram getMemory() {
        return objMemory;
    }
    public int getTotalPrice() {
        return totalPrice;
    }

    // method print
    public void printComputer() {
        System.out.println("BAHASA JAVA\n");
        objProc.printProcie();
        objStorage.printDisk();
        objMemory.printRam();
        System.out.println("Total Price : " + "Rp. " + totalPrice);
    }
}
